File: src/main/java/xyz/qincai/manhunt/ui/GamePhase.java
package xyz.qincai.manhunt.ui;

import xyz.qincai.manhunt.config.ConfigManager;

public enum GamePhase {
    OVERWORLD_PREP("Phase I", "Overworld Prep", "advanced.phase.overworld-prep"),
    NETHER_RUSH("Phase II", "Nether Rush", "advanced.phase.nether-rush"),
    FORTRESS_RUN("Phase III", "Fortress Run", "advanced.phase.fortress-run"),
    BLAZE_ROD_RUN("Phase IV", "Blaze Rod Run", "advanced.phase.blaze-rod-run"),
    BASTION_ROUTE("Phase V", "Bastion Route", "advanced.phase.bastion-route"),
    RETURN_EYES("Phase VI", "Return & Eyes", "advanced.phase.return-eyes"),
    STRONGHOLD_DIVE("Phase VII", "Stronghold Dive", "advanced.phase.stronghold-dive"),
    END_RUSH("Phase VIII", "End Rush", "advanced.phase.end-rush"),
    FINALE("Phase IX", "Finale", "advanced.phase.finale");

    private final String number;
    private final String fallbackName;
    private final String messageKey;

    GamePhase(String number, String fallbackName, String messageKey) {
        this.number = number;
        this.fallbackName = fallbackName;
        this.messageKey = messageKey;
    }

    public String getNumber() {
        return number;
    }

    public String getName() {
        return fallbackName;
    }

    public String getName(ConfigManager configManager) {
        return configManager.getMessage(messageKey);
    }

    public String getDisplay() {
        return number + " \u2014 " + fallbackName;
    }

    public String getDisplay(ConfigManager configManager) {
        return number + " \u2014 " + getName(configManager);
    }
}

File: pumpkin-plugin/game_phase.py
from enum import Enum, auto


class GamePhase(Enum):
    OVERWORLD_PREP = ("Phase I", "Overworld Prep")
    NETHER_RUSH = ("Phase II", "Nether Rush")
    FORTRESS_RUN = ("Phase III", "Fortress Run")
    BLAZE_ROD_RUN = ("Phase IV", "Blaze Rod Run")
    BASTION_ROUTE = ("Phase V", "Bastion Route")
    RETURN_EYES = ("Phase VI", "Return & Eyes")
    STRONGHOLD_DIVE = ("Phase VII", "Stronghold Dive")
    END_RUSH = ("Phase VIII", "End Rush")
    FINALE = ("Phase IX", "Finale")

    def __init__(self, number: str, fallback_name: str):
        self.number = number
        self.fallback_name = fallback_name

    def display(self) -> str:
        return f"{self.number} \u2014 {self.fallback_name}"
